service/src/main/java/bio/terra/workspace/service/resource/controlled/flight/clone/workspace/CloneReferencedResourceStep.java

```diff
@@ -24,20 +24,23 @@
 
 public class CloneReferencedResourceStep implements Step {
 
+  private final AuthenticatedUserRequest userRequest;
   private final SamService samService;
-  private final ResourceDao resourceDao;
+  private final ReferencedResourceService referencedResourceService;
```

Comment:
I am not quite understand why we change the resourceDao to referencedresourceService. how about the controlled  resource?

Reply:
yes this is the meat of the change. the `referencedResourceService` log an activity when creating a referenced resource. Whereas a direct write to the DAO does not log.

This step is called CloneReferencedResourceStep. So it is irrelevant to controlled resource. FWIW, the controlled one is already calling the service.